types/set/nested/write.C

```diff
@@ -0,0 +1,92 @@
+#include <ROOT/RField.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RNTupleWriteOptions.hxx>
+#include <ROOT/RNTupleWriter.hxx>
+
+using ROOT::Experimental::EColumnType;
+using ROOT::Experimental::RField;
+using ROOT::Experimental::RNTupleModel;
+using ROOT::Experimental::RNTupleWriteOptions;
+using ROOT::Experimental::RNTupleWriter;
+
+#include <TSystem.h>
+
+#include <cstdint>
+#include <filesystem>
+#include <memory>
+#include <set>
+#include <string_view>
+
+using Set = std::set<std::set<std::int32_t>>;
+
+static std::shared_ptr<Set> MakeSetField(RNTupleModel &model,
+                                         std::string_view name,
+                                         EColumnType indexType) {
+  auto field = std::make_unique<RField<Set>>(name);
+  field->SetColumnRepresentatives({{indexType}});
+  field->GetSubFields()[0]->SetColumnRepresentatives({{indexType}});
+  model.AddField(std::move(field));
+  return model.GetDefaultEntry().GetPtr<Set>(name);
+}
+
+void write(std::string_view filename = "types.set.nested.root") {
+  if (!std::filesystem::exists("libNestedSet.so")) {
+    throw std::runtime_error("could not find the required ROOT dictionaries, "
+                             "please make sure to run `make` first");
+  }
+
+  gSystem->Load("libNestedSet");
```

Comment:
Another point I just thought about: This will unfortunately not work (out-of-the-box) to execute all macros with a single invocation, like `find . -name write.C -exec root -q -l {} \;`. We can copy `libNestedSet.so` and `NestedSet_rdict.pcm` to the current directory, not sure if there is a way to detect the directory of the macro and automatically load from the right place?

Reply:
>  not sure if there is a way to detect the directory of the macro 

We have `gInterpreter->GetCurrentMacroName()` which give the full pathname of the macro/script file.

Reply:
so
```
TString dirName = gSystem->GetDirName(gInterpreter->GetCurrentMacroName());
```

Reply:
An alternative is to set `LD_LIBRARY_PATH` to the directory containing the dictionary (don't know why I didn't think about that yesterday...). That should be relatively easy to do from (yet-to-be-written) infrastructure to execute all macros, it just needs to replace the `std::filesystem::exists` with just trying `gSystem->Load()` and checking the return value.